docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3359,6 +3370,24 @@ for the table entry "executes" the direct resource appropriately.
         * if **set**: The response includes the counter entry's value read from
           the target.
 
+* `meter_counter_data` field
```

Comment:
To be consistent with the rest of the fields here, have it take effect only when the individual subfields are set ?

Reply:
I feel like @smolkaj has a good point here. Would it make sense to say that if `meter_counter_data` is set, all subfields must also be set?

Reply:
The text below this line mentions about the default configuration which means that all packets are marked green, so would it be ok to have the read request specify other colors ?

Reply:
> The text below this line mentions about the default configuration which means that all packets are marked green, so would it be ok to have the read request specify other colors ?

This is about meter configuration, so I am not sure how this relates to writing / reading the meter's counters?

To clarify @smolkaj's question, consider the case (WriteRequest) where the `meter_counter_data` field is set but not all nested fields are set, e.g. `green` is not set, `yellow` is set to 0 and `red` is not set. What is the expected behavior? One reasonable behavior would be resetting the yellow counter to 0, while leaving the other 2 counters untouched. However, maybe some targets only support resetting all counters simultaneously.

Another possible option:
    * `WriteRequest (INSERT)`
        - if **unset**: The initial value for all 3 counter entries is the default (0).
        - if **set**: Same as **unset** case, subfields are ignored.
    * `WriteRequest (MODIFY)`
        - if **unset**: The 3 counter entry values are not changed.
        - if **set**: All 3 counter entries are reset to 0, subfields are ignored.
    * `ReadRequest`
        * if **unset**: The response does not include the 3 counter entry values.
        * if **set**: The response includes all 3 counter values read from the target.

With this approach, I think we keep things simple. Subfields being set / unset in requests is unrelevant.

The bottom line is that the current version here is to vague because it ignores the fact that there *are* subfields (unlike for normal direct counters).

Reply:
The closes precedence I see is non-key fields being ignored for a delete operation of a table entry.  Will let @antoninbas chip in.

Reply:
@antoninbas I am leaning towards making the INSERT & MODIFY return INVALID_ARGUMENT for non-zero values. That way, we dont change anything and the behavior is well defined. Wdyt ?

Reply:
Sounds fine to me.

Reply:
Updated.